ArcGISRuntimeSDKQt_CppSamples/Analysis/FindServiceAreasForMultipleFacilities/FindServiceAreasForMultipleFacilities.cpp

```diff
@@ -0,0 +1,178 @@
+const QUrl serviceAreaTaskUrl("https://sampleserver6.arcgisonline.com/arcgis/rest/services/NetworkAnalysis/SanDiego/NAServer/ServiceArea");
+}
+
+FindServiceAreasForMultipleFacilities::FindServiceAreasForMultipleFacilities(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_map(new Map(Basemap::lightGrayCanvas(this), this))
+{
+  m_serviceAreasOverlay = new GraphicsOverlay(this);
+
+  // create fill symbols for rendering the results
+  m_fillSymbols.append(new SimpleFillSymbol(SimpleFillSymbolStyle::Solid, QColor(255, 166, 0, 66), this));
+  m_fillSymbols.append(new SimpleFillSymbol(SimpleFillSymbolStyle::Solid, QColor(255, 0, 0, 66), this));
+
+  m_facilitiesTable = new ServiceFeatureTable(url, this);
+  m_facilitiesFeatureLayer = new FeatureLayer(m_facilitiesTable, this);
+
+  // create a symbol to display the facilities
+  PictureMarkerSymbol* facilitiesSymbol = new PictureMarkerSymbol(imageUrl, this);
+  facilitiesSymbol->setWidth(25);
+  facilitiesSymbol->setHeight(25);
+  m_facilitiesFeatureLayer->setRenderer(new SimpleRenderer(facilitiesSymbol, this));
+
+  // add the facilities feature layer to the map
+  m_map->operationalLayers()->append(m_facilitiesFeatureLayer);
+}
+
+FindServiceAreasForMultipleFacilities::~FindServiceAreasForMultipleFacilities() = default;
+
+void FindServiceAreasForMultipleFacilities::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<FindServiceAreasForMultipleFacilities>("Esri.Samples", 1, 0, "FindServiceAreasForMultipleFacilitiesSample");
+}
+
+MapQuickView* FindServiceAreasForMultipleFacilities::mapView() const
+{
+  return m_mapView;
+}
+
+bool FindServiceAreasForMultipleFacilities::taskRunning() const
+{
+  return m_taskRunning;
+}
+
+// Set the view (created in QML)
+void FindServiceAreasForMultipleFacilities::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  m_mapView->graphicsOverlays()->append(m_serviceAreasOverlay);
+
+  connect(m_facilitiesFeatureLayer, &FeatureLayer::doneLoading, this, [this](Error loadError){
+    if (!loadError.isEmpty())
+    {
+      qDebug() << loadError.message() << loadError.additionalMessage();
+      return;
+    }
+
+    if (m_facilitiesFeatureLayer->loadStatus() == LoadStatus::Loaded)
+    {
+      // zoom to the full extent of the feature layer
+      m_mapView->setViewpointGeometry(m_facilitiesFeatureLayer->fullExtent(), 100);
+    }
+  });
+
+  emit mapViewChanged();
+}
+
+void FindServiceAreasForMultipleFacilities::findServiceAreas()
+{
+  // start showing indicator
+  m_taskRunning = true;
+  emit taskRunningChanged();
+
+  m_serviceAreaTask = new ServiceAreaTask(serviceAreaTaskUrl, this);
+  m_serviceAreaTask->load();
+
+  // once service area task is done loading, create default parameters
+  connect(m_serviceAreaTask, &ServiceAreaTask::doneLoading, this, [this](Error loadError)
```

Comment:
again - move this before the load call. I think the whole setup for the task should be moved out of this function so you have two separate calls - 1 to set everything up and 1 to actually do the work to find service areas.